tests/lib/tools/systemd-state

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: create-and-start-unit <unit-name> <unit-args> [--overrides <conf>] [--persistent]"
+    echo "       stop-and-remove-unit <unit-name>"
+    echo "       wait-for-unit <unit-name>"
+    echo "       stop-units <unit-names>"
+    echo "       active-snapd-units"
+    echo ""
+    echo "Manage systemd service units and retrieve information about existing units."
+    echo ""
+    echo "COMMANDS:"
+    echo "  create-and-start-unit:   creates and starts systemd unit. In case it is set as persistent the unit survive reboots. Also it allows to override the configuration."
+    echo "  stop-and-remove-unit:    stops and removes a service unit"
+    echo "  wait-for-unit:           waits until a service unit is active"
+    echo "  stop-units:              stops a list of service units"
+    echo "  active-snapd-units:      retrieve a list ov active snapd units"
+}
+
+# Use like systemd_create_and_start_unit "fakestore" "$(which fakestore) -start -dir $top_dir -addr localhost:11028 $@"
+systemd_create_and_start_unit() {
+    printf '[Unit]\nDescription=Support for test %s\n[Service]\nType=simple\nExecStart=%s\n' "${SPREAD_JOB:-unknown}" "$2" > "/run/systemd/system/$1.service"
+    if [ -n "${3:-}" ]; then
+        echo "Environment=$3" >> "/run/systemd/system/$1.service"
+    fi
+    systemctl daemon-reload
+    systemctl start "$1"
+}
+
+# Use like systemd_create_and_start_persistent_unit "name" "start" "[Unit]\nAfter=foo"
+systemd_create_and_start_persistent_unit() {
```

Comment:
Same suggestion, I would suggest not having this command and have users mkdir / echo the unit files directly.